lib/blocktree/blocktree.go

```diff
@@ -68,6 +70,50 @@ func (bt *BlockTree) GenesisHash() Hash {
 	return bt.head.hash
 }
 
+// GetAllHashesForParentDepth will get all hashes with the same dept for a given header
+func (bt *BlockTree) GetAllHashesForParentDepth(header *types.Header) (map[common.Hash]*big.Int, error) {
+	hashes := map[common.Hash]*big.Int{}
+
+	parent := bt.getNode(header.ParentHash)
+	if parent == nil {
+		err := fmt.Errorf("cannot find parent block in blocktree")
+		log.Error("Could not getNode for ParentHash", "header.ParentHash", header.ParentHash, "err", err)
+		return hashes, err
+	}
+
+	for _, child := range bt.head.children {
```

Comment:
how does this work? it isn't recursive so it won't go past depth 1 of the tree

Reply:
is it ? I was thinking that the bt.head was the genesis, and all its children are then all the blocks following it. Do you have any other suggestions ?

Reply:
no, the children aren't the blocks that follow it sequentially, the children are the blocks that all have the block as the parent hash. the head isn't necessarily the genesis, right now it is only because we don't have finality. I will take a look at everything and see how this works

Reply:
right, that makes more sense...you are right, I updated it a bit to reflect this line of thinking, could you please check again ? :)